OpenTaiko/src/Stages/11.Heya/CStageHeya.cs

```diff
@@ -557,8 +557,10 @@ public override int Draw() {
 			if (this.tMove(-1)) {
 				OpenTaiko.Skin.soundChangeSFX.tPlay();
 			}
-		} else if (OpenTaiko.InputManager.Keyboard.KeyPressed((int)SlimDXKeys.Key.Return) ||
-				   OpenTaiko.Pad.bPressed(EInstrumentPad.Drums, EPad.Decide)) {
+		}
+		else if (iCurrentMenu != CurrentMenu.Name && OpenTaiko.InputManager.Keyboard.KeyPressed((int)SlimDXKeys.Key.Return) ||
+				   OpenTaiko.Pad.bPressed(EInstrumentPad.Drums, EPad.Decide))

```

Comment:
`p && q || r` ↔ `(p && q) || r`
Return and Decide has different behaviors. Is this intended?

Reply:
`Return` is another term for the keyboard button `Enter`, which isn't the same as Decide, which is an action that you add keybinds to.

Reply:
For why Decide/Cancel is missing in some places, Decide/Cancel can contain keyboard inputs which a user may also use to type letters into their name. So the Decide/Cancel action is now excluded if the current menu is set to Name, and will only take Return/Escape instead.

Reply:
Are you sure you didn't forgot parenthesis here? It would trigger is Decide is pressed within the Name menu

Reply:
My bad, I understand the question now

fixed